financial-connections/src/main/java/com/stripe/android/financialconnections/FinancialConnectionsSheet.kt

```diff
@@ -42,36 +43,14 @@ class FinancialConnectionsSheet internal constructor(
     @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
     @Parcelize
     data class ElementsSessionContext(
-        val initializationMode: InitializationMode,
         val amount: Long?,
         val currency: String?,
         val linkMode: LinkMode?,
         val billingDetails: BillingDetails?,
         val prefillDetails: PrefillDetails,
+        val incentiveEligibilitySession: IncentiveEligibilitySession?
     ) : Parcelable {
 
-        val paymentIntentId: String?
-            get() = (initializationMode as? InitializationMode.PaymentIntent)?.paymentIntentId
-
-        val setupIntentId: String?
-            get() = (initializationMode as? InitializationMode.SetupIntent)?.setupIntentId
-
-        @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
-        sealed interface InitializationMode : Parcelable {
```

Comment:
Replaced `InitializationMode` with `IncentiveEligibilitySession` to make things a little clearer.

Reply:
Much clearer indeed!

Reply:
Also, `InitializationMode` used to not be nullable, but `IncentiveEligibilitySession` is (which makes sense), curious why it wasn't nullable before?

Reply:
We now only pass an `IncentiveEligibilitySession` if the session if eligible for an incentive, and we pass `null` if it’s not.